### PR TITLE
Fix Copy cite command should respect preferences

### LINKED ISSUE #10615: Copy cite command should respect preferences
https://github.com/JabRef/jabref/pull/10303 introduced a new cite command:

image

It is parsed at `org.jabref.gui.push.AbstractPushToApplication#dissectCiteCommand`.

However, in `org.jabref.gui.edit.CopyMoreAction#copyCiteKey`, this pattern is not used. 

This leads to `\cite{key1,key2}{Khalaf2007}` for key `Khalaf2007`.

284869839-992b6afe-6efc-4a6d-811f-0641c392a50d

## Implementation

1. Rename `org.jabref.logic.citationkeypattern.CitationKeyPatternPreferences` to `CitationKeyGenerationPreferences`.
1. Modify `org.jabref.preferences.ExternalApplicationsPreferences#getCiteCommand` to return a `record` having `prefix, delimiter, suffix` as string. There, the value is constructed based on the code at org.jabref.gui.push.AbstractPushToApplication#dissectCiteCommand. - No need to introduce caching there. That could be done as next step. (This code does not run for a long time, so no performance issues expected). Name the `record` `CitationCommandStringPreferences`.
1. Add test cases testing (wrong) preference `\cite`. It should work nevertheless. It should use `,` as delimiter and `{` as prefix, and `}` as suffix (as fallback if nothing was provided).  Adapt the code of `CitationKeyGenerationPreferences` accordingly. If this is successfully done, https://github.com/JabRef/jabref/issues/10568 is fixed.
1. Adapt `getCitePrefix()`, ` getDelimiter()`, and `getCiteSuffix()` to use the introduced `CitationStringPreferences`.
1. Remove `org.jabref.gui.push.AbstractPushToApplication#dissectCiteCommand`.
1. <s>Remove `org.jabref.gui.push.PushToApplication#getDelimiter`
   - Remove method
   - Change parameter at org.jabref.gui.push.PushToApplicationCommand#pushEntries to use `CitationCommandStringPreferences`s delimiter</s> This is wrong; since application's delimiter has higher precedence than the one of the citaton command.
1. Adapt `org.jabref.gui.edit.CopyMoreAction#copyCiteKey` to use the new preference

### PR DESCRIPTION
**Pull request by Siedlerchr**
Fixes #10615
Supersedes https://github.com/JabRef/jabref/pull/10690

### Mandatory checks

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by koppor** [comment]
I'm working on it - especially that `\cite` as preference works. Reason: We should not force all users to adapt their preferences. JabRef should keep working. See https://github.com/JabRef/jabref/issues/10568#issuecomment-1776929318 for a discussion.

**Comment by Siedlerchr** [comment]
I'm confused. The default cite command is still \cite

**Comment by koppor** [comment]
The issue also happens if the user enters something wrong in the preference field. Currently, `key1key2` is then used. Which can be done better. I am working on it.

I will output an info

    INFO: Wrong indexes -1 -1 for completeCiteCommand \cite. Using default delimiter and suffix.

`\cite` then automatically turns into `\cite{key1,key2}`

**Comment by koppor** [comment]
image

Registry setting `Computer\HKEY_CURRENT_USER\SOFTWARE\JavaSoft\Prefs\org\jabref` -> `cite/Command`, content `\cite` will be magically converted by that parsing, too.

**Comment by koppor** [comment]
I went through the items of https://github.com/JabRef/jabref/issues/10615 and worked on all of them (e.g., adding more test cases). Should be fine now 😅

File: src/main/java/org/jabref/gui/preferences/external/ExternalTabViewModel.java [hunk 1]
```diff
@@ -39,7 +40,7 @@ public class ExternalTabViewModel implements PreferenceTabViewModel {
     private final BooleanProperty autoOpenAttachedFoldersProperty = new SimpleBooleanProperty();
     private final ListProperty<PushToApplication> pushToApplicationsListProperty = new SimpleListProperty<>();
     private final ObjectProperty<PushToApplication> selectedPushToApplicationProperty = new SimpleObjectProperty<>();
-    private final StringProperty citeCommandProperty = new SimpleStringProperty("");
+    private final StringProperty citeCommandProperty = new SimpleStringProperty();
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/preferences/external/ExternalTabViewModel.java:43]: Better use empty string to avoid possible npe

Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/preferences/external/ExternalTabViewModel.java:43]:
```suggestion
    private final StringProperty citeCommandProperty = new SimpleStringProperty("");
```

Comment (koppor) [reply src/main/java/org/jabref/gui/preferences/external/ExternalTabViewModel.java]: Please tell this the Siedlerchr of 3 hours ago :p

**Review by koppor — DISMISSED** [review]

**Review by koppor — APPROVED** [review]

**Pull request merged**